### PR TITLE
Add code coverage github action

### LINKED ISSUE #318: Add code coverage reporting
Follow up from https://github.com/open-telemetry/opentelemetry-rust/issues/143

### PR DESCRIPTION
**Pull request by jtescher**
Pull out nightly tests into a new CI run that includes test coverage using `grcov` and `codecov`

Resolves #318

### CODE REVIEW

**Review by TommyCpp — APPROVED** [review]
LGTM :+1:

**Comment by jtescher** [comment]
Want to see if I can get the otlp/prometheus exporters in here too, something about grpcio is having issues with the coverage flags. If not can get things started here and improve in subsequent PRs.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,6 +49,33 @@ jobs:
...
+        args: --all-features --all
+    - uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --all-features --no-fail-fast -p opentelemetry -p opentelemetry-semantic-conventions -p opentelemetry-jaeger
+      env:
+        CARGO_INCREMENTAL: '0'
+        RUSTFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests'
+        RUSTDOCFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests'
+    - uses: actions-rs/grcov@v0.1
```
Comment (drexler) [inline .github/workflows/ci.yml:80]: We could use the latest version of grcov here - v0.1.5

Comment (jtescher) [reply .github/workflows/ci.yml:80]: Looks like this is the latest? (v0.1 is pointing at same as v0.1.6) https://github.com/actions-rs/grcov/tags

**Comment by jtescher** [comment]
Ok going to merge this for now to get a coverage baseline.

**Pull request merged**